cpp/dolfinx/io/xdmf_mesh.cpp

```diff
@@ -277,10 +259,26 @@ xdmf_mesh::read_mesh_data(MPI_Comm comm, const hid_t h5_id,
   // Read geometry data
   const std::vector<double> geometry_data
```

Comment:
Would be nice to change the return type at some point.

Reply:
I will be templating MeshTags IO in later PRs.

Reply:
I had in mind using Eigen to get multi-dimensional array.